hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/stores/WritableTssStore.java

```diff
@@ -57,44 +71,100 @@ public WritableTssStore(@NonNull final WritableStates states) {
+     * @param txBody            the {@link TssVoteTransactionBody} for the given roster to be persisted
+     */
     public void put(@NonNull final TssVoteMapKey tssVoteMapKey, @NonNull final TssVoteTransactionBody txBody) {
         requireNonNull(tssVoteMapKey);
         requireNonNull(txBody);
         tssVoteState.put(tssVoteMapKey, txBody);
     }
 
+    /**
+     * Persists a new {@link EntityNumber} with given {@link TssEncryptionKeyTransactionBody} into the state.
+     *
+     * @param entityNumber      the corresponding Node ID to the {@link TssEncryptionKeyTransactionBody} to be persisted
+     * @param txBody            the {@link TssEncryptionKeyTransactionBody} for the given node to be persisted
+     */
     public void put(@NonNull final EntityNumber entityNumber, @NonNull final TssEncryptionKeyTransactionBody txBody) {
         requireNonNull(entityNumber);
         requireNonNull(txBody);
         tssEncryptionKeyState.put(entityNumber, txBody);
     }
 
+    /**
+     * Persists a new {@link TssStatus} for the current Tss state.
+     *
+     * @param tssStatus the {@link TssStatus} to be persisted
+     */
     public void put(@NonNull final TssStatus tssStatus) {
         requireNonNull(tssStatus);
         tssStatusState.put(tssStatus);
     }
 
+    /**
+     * Removes a {@link TssMessageTransactionBody} from the state.
+     *
+     * @param tssMessageMapKey for which the {@link TssMessageTransactionBody} to be removed.
+     */
     public void remove(@NonNull final TssMessageMapKey tssMessageMapKey) {
         requireNonNull(tssMessageMapKey);
         tssMessageState.remove(tssMessageMapKey);
     }
 
+    /**
+     * Removes a {@link TssVoteTransactionBody} from the state.
+     *
+     * @param tssVoteMapKey for which the {@link TssVoteTransactionBody} to be removed.
+     */
     public void remove(@NonNull final TssVoteMapKey tssVoteMapKey) {
         requireNonNull(tssVoteMapKey);
         tssVoteState.remove(tssVoteMapKey);
     }
 
+    /**
+     * Removes a {@link TssEncryptionKeyTransactionBody} from the state.
+     *
+     * @param entityNumber the Node for which the {@link TssEncryptionKeyTransactionBody} to be removed.
+     */
     public void remove(@NonNull final EntityNumber entityNumber) {
         requireNonNull(entityNumber);
         tssEncryptionKeyState.remove(entityNumber);
     }
 
+    /**
+     * Removes EntityNumber (Node ID) from the {@link TssEncryptionKeyTransactionBody} map, but only if
+     * the Node ID is present in neither the active roster's and the candidate roster's entries. {@link RosterEntry}
+     *
+     * @param rostersEntriesNodeIds contains the non-duplicate Node IDs of current active and candidate rosters entries
+     */
+    public void removeIfNotPresent(@NonNull final Set<EntityNumber> rostersEntriesNodeIds) {
+        requireNonNull(rostersEntriesNodeIds);
+        tssEncryptionKeyState.keys().forEachRemaining(entityNumber -> {
```

Comment:
Iterating through all keys in state in general is not recommended, can we avoid that?

Reply:
I agree, not a good practice indeed but couldn't think of other way to remove any TssEncryptionTx bodies from state that are not present in the current rosters. Also, I don't think that in any scenario we end up with enormous amount of TssEncryptionTx bodies already in state to iterate over, especially when the clean up is performed on every tss state change. Thoughts? @MiroslavGatsanoga @derektriley @tinker-michaelj

Reply:
Since there shouldn't be a lot of keys to iterate over, this seems ok. We can keep an eye on it